integration-tests/src/test/java/tachyon/master/MasterClientAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,188 @@
+        AuthType.SIMPLE.getAuthName());
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    testAuthenticationOperation("/file-simple");
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+    System.clearProperty(Constants.TACHYON_SECURITY_AUTHENTICATION);
+  }
+
+  @Test
+  public void customAuthenticationOpenCloseTest() throws Exception {
+    //custom authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.CUSTOM.getAuthName());
+    //custom authenticationProvider configure
+    System.setProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+
+    /**
+     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
+     * Tachyon Master.
+     */
+    LoginUser.setLoginUser(LoginUser.createUserForTesting("tachyon"));
+
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    testAuthenticationOperation("/file-custom");
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+    System.clearProperty(Constants.TACHYON_SECURITY_AUTHENTICATION);
+    System.clearProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS);
+  }
+
+  @Test
+  public void customAuthenticationDenyConnectTest() throws Exception {
+    //custom authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.CUSTOM.getAuthName());
+    //custom authenticationProvider configure
+    System.setProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+    /**
+     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
+     * Tachyon Master.
+     */
+    LoginUser.setLoginUser(LoginUser.createUserForTesting("tachyon"));
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    mMasterInfo = mLocalTachyonCluster.getMasterInfo();
+    // Using no-tachyon as loginUser to connect to Master, the IOException will be thrown
+    try {
+      mThrown.expect(IOException.class);
+      LoginUser.setLoginUser(LoginUser.createUserForTesting("no-tachyon"));
+      MasterClient masterClient =
+          new MasterClient(mMasterInfo.getMasterAddress(),
+              mExecutorService, mLocalTachyonCluster.getMasterTachyonConf());
+      Assert.assertFalse(masterClient.isConnected());
+      masterClient.connect();
+    } finally {
+      mLocalTachyonCluster.stop();
+      System.clearProperty(Constants.TACHYON_SECURITY_AUTHENTICATION);
+      System.clearProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS);
+    }
+  }
+
+  /**
+   * Test Tachyon client connects or disconnects to the Master. When the client connects
+   * successfully to the Master, it can successfully create file or not.
+   * @param filename
+   * @throws Exception
+   */
+  private void testAuthenticationOperation(String filename) throws Exception {
```

Comment:
could you name this as `authenticationOperationTest` to stay consistent with the rest of the unittests?